crates/samples/services/thread/src/main.rs

```diff
@@ -1,44 +1,54 @@
 use windows_services::*;
 use windows_threading::*;
 
+use std::sync::{Arc, RwLock};
+
 fn main() {
-    let pool = Pool::new();
+    let pool: Pool = Pool::new();
     pool.set_thread_limits(1, 1);
 
-    Service::new()
+    let service_original = Arc::new(RwLock::new(Service::new()));
+    let service = Arc::clone(&service_original);
+    service_original
```

Comment:
This seems overly complicated. The original `Service::run` terminated the process upon completion so that the `run` method was `-> !` and the lifetime was much simpler. This was changed in #3662 to make it possible to test `Service` but perhaps that was a mistake.